src/bindings/python/src/pyopenvino/core/core.cpp

```diff
@@ -552,40 +552,49 @@ void regclass_Core(py::module m) {
-                char* buffer;
-                Py_ssize_t length;
-                PYBIND11_BYTES_AS_STRING_AND_SIZE(data.ptr(), &buffer, &length);
-                _stream.write(buffer, length);
-                _stream.close();
-            } else {
-                OPENVINO_THROW("Failed to open temporary file for model stream");
-            }
-
             ov::CompiledModel result;
-            std::fstream _fstream(filename, std::ios::in | std::ios::binary);
-            if (_fstream.is_open()) {
-                py::gil_scoped_release release;
-                result = self.import_model(_fstream, device_name, _properties);
-                _fstream.close();
-                if (std::remove(filename.c_str()) != 0) {
-                    const std::string abs_path =
-                        py::module_::import("os").attr("getcwd")().cast<std::string>() + "/" + filename;
-                    const std::string warning_message = "Temporary file " + abs_path + " failed to delete!";
-                    PyErr_WarnEx(PyExc_RuntimeWarning, warning_message.c_str(), 1);
+            // std::stringstream cannot handle streams > 2GB, in that case use std::fstream
+            if (size_in_gb > 2) {
+                std::random_device rd;
+                std::mt19937 gen(rd());
+                std::uniform_int_distribution<> distr(1000, 9999);
+                std::string filename = "model_stream_" + std::to_string(distr(gen)) + ".txt";
+                std::fstream _stream(filename, std::ios::out | std::ios::binary);
+                if (_stream.is_open()) {
+                    const py::bytes data = model_stream.attr("read")();
+                    // convert the Python bytes object to C++ string
+                    char* buffer;
+                    Py_ssize_t length;
+                    PYBIND11_BYTES_AS_STRING_AND_SIZE(data.ptr(), &buffer, &length);
+                    _stream.write(buffer, length);
+                    _stream.close();
+                } else {
+                    OPENVINO_THROW("Failed to open temporary file for model stream");
+                }
+                std::fstream _fstream(filename, std::ios::in | std::ios::binary);
+                if (_fstream.is_open()) {
+                    py::gil_scoped_release release;
+                    result = self.import_model(_fstream, device_name, _properties);
+                    _fstream.close();
+                    if (std::remove(filename.c_str()) != 0) {
+                        const std::string abs_path =
+                            py::module_::import("os").attr("getcwd")().cast<std::string>() + "/" + filename;
+                        const std::string warning_message = "Temporary file " + abs_path + " failed to delete!";
+                        PyErr_WarnEx(PyExc_RuntimeWarning, warning_message.c_str(), 1);
+                    }
+                } else {
+                    OPENVINO_THROW("Failed to open temporary file for model stream");
```

Comment:
OPENVINO_ASSRT(fstream.is_open(), "Failed to open temporary file for model stream")

to minimize a number of if / else blocks.

Reply:
Good idea, applied.